### PR TITLE
Add Error when Loading Lambs only on Clients but not on the Server

### PR DESCRIPTION
**Pull request by jokoho48**
### When merged this pull request will:

1. *Title*

### CODE REVIEW

File: addons/main/XEH_preInit.sqf [hunk 1]
```diff
@@ -11,4 +11,29 @@ GVAR(shareHandlers) = [];
 
 GVAR(blockSuppressionModelCache) = false call CBA_fnc_createNamespace;
 
+if (isServer) then {
+    GVAR(versionLoadedOnServer) = QUOTE(VERSION_STR);
+    publicVariable QGVAR(versionLoadedOnServer);
+} else {
+    0 spawn {
+        if (GVAR(versionLoadedOnServer) isEqualTo QUOTE(VERSION_STR)) exitWith {};
```
Comment (diwako) [inline addons/main/XEH_preInit.sqf:19]: Nil check for `GVAR(versionLoadedOnServer)` after mission start.

**Review by diwako — CHANGES_REQUESTED** [review]

File: addons/main/XEH_preInit.sqf [hunk 2]
```diff
@@ -11,4 +11,29 @@ GVAR(shareHandlers) = [];
...
+            "Lambs Danger Version mismatch Error.<br/>Client Version: %1<br/>Server Version: %2",
+            QUOTE(VERSION_STR),
+            GVAR(versionLoadedOnServer)
+        ];
+        private _error = format [
+            "Lambs Danger Version mismatch Error. Client Version: %1 Server Version: %2",
+            QUOTE(VERSION_STR),
+            GVAR(versionLoadedOnServer)
+        ];
+        while {true} do {
```
Comment (diwako) [inline addons/main/XEH_preInit.sqf:30]: you sure you want to spam that every second?

Comment (jokoho48) [reply addons/main/XEH_preInit.sqf:30]: yes

File: addons/main/XEH_preInit.sqf [hunk 3]
```diff
@@ -11,4 +11,30 @@ GVAR(shareHandlers) = [];
 
 GVAR(blockSuppressionModelCache) = false call CBA_fnc_createNamespace;
 
+if (isServer) then {
+    GVAR(versionLoadedOnServer) = QUOTE(VERSION_STR);
+    publicVariable QGVAR(versionLoadedOnServer);
+} else {
+    0 spawn {
+        waitUntil {time > 0};
+        if (!isNil QGVAR(versionLoadedOnServer) && GVAR(versionLoadedOnServer) isEqualTo QUOTE(VERSION_STR)) exitWith {};
```
Comment (veteran29) [inline addons/main/XEH_preInit.sqf:20]: Needs lazy eval, otherwise it's still comparing nil to string

Comment (diwako) [reply addons/main/XEH_preInit.sqf:20]: I blame Joko

Comment (jokoho48) [reply addons/main/XEH_preInit.sqf:20]: fixed on master ^^ thanks. but stupid SQF 🤣

Comment (rautamiekka) [reply addons/main/XEH_preInit.sqf:20]: > stupid SQF

Amen to that. _Won't be getting much better ..._

**Pull request merged**